### ISSUE TITLE
NAP is creating NodeClaims and VMs but not registering Nodes to the cluster

### ISSUE THREAD

**Issue by WillEdMatsypie** [Azure/karpenter-provider-azure#248]
### Version

__Karpenter Overlay Version:__ N/A - we are using managed NAP

__Kubernetes Version:__ v1.28.5

### Expected Behavior

Karpenter should create NodeClaims and provision VMs which then register to the cluster and are capable of scheduling Pods. 

### Actual Behavior

Pods go into pending state and create a NodeClaim with Karpenter i.e. 
![image](https://github.com/Azure/karpenter-provider-azure/assets/43652033/51a3da71-bd5c-4e9d-aec9-694d2177e6b0)

The NodeClaims then create and provision a VM but never reach Ready state  
![image](https://github.com/Azure/karpenter-provider-azure/assets/43652033/b1023aa1-ed6c-465e-9cc6-3f07969b252c)

The reason given in the Claim is NodeNotFound: 
![image](https://github.com/Azure/karpenter-provider-azure/assets/43652033/2444cfb0-1983-4a6a-9858-f4be1ff990fe)

We can see the VMs are up and running in the account: 
![image](https://github.com/Azure/karpenter-provider-azure/assets/43652033/febc48a8-f03d-4905-892a-2a800da71511)

### Steps to Reproduce the Problem

N/A - this is a managed service. In our cluster this was working until yesterday afternoon and then Karpenter stopped being able to register Nodes with the cluster altogether.

### Resource Specs and Logs

N/A - issue is with managed NAP not registering nodes to the cluster, no logs we can see outside of those shared in the "Actual behaviour" above.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by Bryce-Soghigian** [comment]
I suspect its related to https://github.com/Azure/karpenter-provider-azure/issues/247

**Comment by Bryce-Soghigian** [comment]
Will take a look today

**Comment by Bryce-Soghigian** [comment]
I took a look at your cluster. 

2024-04-01T00:37:27.778Z we see that a NodeImageUpgrade occurred on the agentpool, which triggered #247   where we see the bootstrap token isn't rotated when one of the agentpools is upgraded. We can fix this by doing `az aks update -g rg -n clusterName`, which simply reapplys the token and passes that configuration to NAP. 

I have mitigated the problem for you by running the az aks update -g rg -n clusterName with a powerful azure employee command. 

The fix for this on the nap side is currently rolling out, but this should fix things
